pkg/providers/subnet/suite_test.go

```diff
@@ -114,6 +114,71 @@ var _ = Describe("SubnetProvider", func() {
 					AvailabilityZoneId:      lo.ToPtr("tstz1-1a"),
 					AvailableIpAddressCount: lo.ToPtr[int32](100),
 					VpcId:                   lo.ToPtr("vpc-test1"),
+					CidrBlock:               lo.ToPtr("10.0.1.0/24"),
+				},
+			}, subnets)
+		})
+		It("should discover subnet by CidrBlock", func() {
```

Comment:
Can you add integration tests for this field? https://github.com/aws/karpenter-provider-aws/blob/main/test/suites/integration/subnet_test.go